sparse/tests/test_coo.py

```diff
@@ -250,7 +250,128 @@ def test_elemwise_trinary(func, shape):
     y = ys.todense()
     z = zs.todense()
 
-    assert_eq(sparse.elemwise(func, xs, ys, zs), func(x, y, z))
+    fs = sparse.elemwise(func, xs, ys, zs)
+    assert isinstance(fs, COO)
+
+    assert_eq(fs, func(x, y, z))
+
+
+@pytest.mark.parametrize('shapes, func', [
+    ([
+         (2,),
+         (3, 2),
+         (4, 3, 2),
+     ], lambda x, y, z: (x + y) * z),
```

Comment:
Consider doing a full cross-product of shapes and functions here.

Reply:
Done!